src/server/scripts/Spells/spell_generic.cpp

```diff
@@ -4460,6 +4460,33 @@ class spell_gen_impatient_mind : public SpellScriptLoader
         }
 };
 
+// 180457 - Unholy Resurrection.
+class spell_gen_rasthe_unholy_resurrection : public SpellScriptLoader
+{
+public:
+    spell_gen_rasthe_unholy_resurrection() : SpellScriptLoader("spell_gen_rasthe_unholy_resurrection") { }
+
+    class spell_gen_rasthe_unholy_resurrection_SpellScript : public SpellScript
+    {
+        PrepareSpellScript(spell_gen_rasthe_unholy_resurrection_SpellScript);
+
+        void HandleScript(SpellEffIndex /*effIndex*/)
+        {
+             if (Unit* target = GetHitUnit())
```

Comment:
Sorry, but i disagree.
"Working" is not enough.... it should be implemented more efficiently possible.
If i'm not wrong, ```OnEffectHitTarget``` hook is called when spell hit target... so always GetHitUnit will not be null in this hook when spell only hit units. Can be null if hit objects. 
In this case, he can only make a variable cache, without condition.
like: 
```cpp
Unit* target = GetHitUnit();
target->SetHealth(target->CountPctFromMaxHealth(25));
```